server/src/main/java/org/elasticsearch/action/delete/DeleteResponse.java

```diff
@@ -40,7 +40,11 @@ public DeleteResponse() {
     }
 
     public DeleteResponse(ShardId shardId, String type, String id, long seqNo, long primaryTerm, long version, boolean found) {
-        super(shardId, type, id, seqNo, primaryTerm, version, found ? Result.DELETED : Result.NOT_FOUND);
+        this(shardId, type, id, seqNo, primaryTerm, version, found ? Result.DELETED : Result.NOT_FOUND);
+    }
+
+    private DeleteResponse(ShardId shardId, String type, String id, long seqNo, long primaryTerm, long version, Result result) {
+        super(shardId, type, id, seqNo, primaryTerm, version, result);
```

Comment:
I think it might be worth asserting that `Result` is one of `DELETED or NOT_FOUND`.